spec/urn_spec.rb

```diff
@@ -2,51 +2,61 @@
 require 'urn'
 
 RSpec.describe URN do
-  describe '#valid?' do
-    it 'returns true if it is valid' do
-      expect(described_class.new('urn:namespace:specificstring')).to be_valid
+  describe 'Kernel.URN' do
+    it 'returns a URN if it is valid' do
+      expect(URN('urn:namespace:specificstring')).to be_kind_of(described_class)
     end
 
-    it 'returns true if namespace includes urn' do
-      expect(described_class.new('urn:urnnamespace:specificstring')).to be_valid
+    it 'raise InvalidURNError if it is not valid' do
+      expect { URN('urn:urn:1234') }.to raise_error(described_class::InvalidURNError, 'bad URN(is not URN?): urn:urn:1234')
+    end
+  end
+
+  describe '#initialize' do
+    it 'returns a URN if it is valid' do
+      expect(described_class.new('urn:namespace:specificstring')).to be_kind_of(described_class)
+    end
+
+    it 'raise InvalidURNError if it is not valid' do
+      expect { described_class.new('urn:urn:1234') }.to raise_error(described_class::InvalidURNError, 'bad URN(is not URN?): urn:urn:1234')
     end
 
-    it 'returns false if it does not start with urn' do
-      expect(described_class.new('not-urn:namespace:specificstring')).not_to be_valid
+    it 'returns a URN if namespace includes urn' do
+      expect(described_class.new('urn:urnnamespace:specificstring')).to be_kind_of(described_class)
     end
 
-    it 'returns false if namespace is urn' do
-      expect(described_class.new('urn:urn:specificstring')).not_to be_valid
+    it 'returns error if it does not start with urn' do
```

Comment:
Would it more accurate to say this _raises_ an error rather than _returning_ one?